src/utils/errors.js

```diff
@@ -7,57 +7,91 @@
+    viewMethodMayNotBeOverridden: function(methodName, debugName) {
+      return `View.${methodName} may not be overridden` + (debugName ? ` for view "${debugName}"` : '');
+    },
+    componentFunctionMayNotBeCalledDirectly: (fn) => `Component.${fn} may not be called directly`,
+    cannotOverwriteComponentMethod: (fn) => `Cannot overwrite component method: ${fn}`,
+    unableToParseToAValidValueMustMatchJSONFormat: (value) => `Unable to parse "${value}" to a valid value. Input must match JSON format`,
+    widgetTypeHasNoTemplateToStringFunctionFallingBackToDOM: (name) => `Widget type: ${name} has no templateToString function, falling back to DOM`,
+    objectDoesNotMeetExpectedEventSpec: () => 'Object does not meet expected event spec',
+    warningNoPartialRegisteredWithTheName: (partialName) => `Warning: no partial registered with the name ${partialName}`,
+    doubleCurlyInterpolatorsCannotBeInAttributes: () => 'Double curly interpolators cannot be in attributes',
+    tagsImproperlyPairedClosing: (tagName, openID, id) => `${tagName} tags improperly paired, closing ${openID} with close tag from ${id}`,
+    closingElementWhenTheStackWasEmpty: (id) => `Closing element ${id} when the stack was empty`,
+    computedPropertiesAreNowDeprecatedPleaseChange: (name) => `Computed properties are now deprecated and will be removed soon. Please change "${name}" to a derived property`
+  },
+  error: {
+    unableToLaunchDebugPanel: () => 'Unable to launch debug panel. You may need to allow the popup or run "window.launchDebugger()" from your console'
+  },
+  exception: {
+  }
+};
+
+/**
+ * Returns a logger call with the provided message and type
+ * @param  {function} message  function that returns our message
+ * @param  {string} type The type of log to make (warning, error, exception, etc...)
+ * @return {function}      Function that calls logger with the provided message and type
+ */
+function loggerize(message, type) {
+  return function() {
+    logger[type](message.apply(message, arguments));
+    // Return the message in case it's needed. (I.E. for utils.alert)
+    return message.apply(message, arguments);
+  };
+}
+
 /**
- * Extends a generic error message with custom information, i.e. implementation-specific hints
- * @param  {string} error The name of the error function to extend
+ * Extends a generic error message with custom information (i.e. implementation-specific hints)
+ * and then adds the loggerized version of the extended message to module.exports.
+ * @param  {string} errorName The name of the error function to extend
  * @param  {string} customMsg The custom message to append to the generic error message.
  */
-function extend(error, customMsg) {
-  if (!this.hasOwnProperty(error)) {
-    logger.warn(`Tried to extend a nonexistent error message: ${error}`);
-    return;
+function extend(errorName, customMsg) {
+  // Find our error message
+  for (let type in messages) {
+    if (messages[type].hasOwnProperty(errorName)) {
```

Comment:
`hasOwnProperty` isn't necessary since it's plain object; let's remove since it's a bit of a performance hit.